public/pages/createDetector/containers/DataSource/DataSource.tsx

```diff
@@ -131,11 +155,25 @@ function DataSource(props: DataFilterProps) {
             isInvalid={isInvalid(field.name, form)}
             error={getError(field.name, form)}
           >
-            <EuiSelect
-              {...field}
+            <EuiComboBox
               id="timeField"
               placeholder="Find timestamp"
               options={timeStampFieldOptions}
+              onSearchChange={handleSearchChange}
+              onCreateOption={(createdOption: string) => {
+                const normalizedOptions = createdOption.trim();
+                if (!normalizedOptions) return;
+                form.setFieldValue('timeField', normalizedOptions);
+              }}
```

Comment:
why allow user to create option? what if user inputs invalid timestamp field?

Reply:
Similar questions: what if user input invalid filed in custom expression of filter and feature definition.
If user input invalid filed, like they input a filed which is string not timestamp, or the filed not exists, the detector just can't get any data to run. 
A better way is we can pop up some warning message and block create detector with invalid parameters. 
Amit is working on auto AD project. We can discuss that design, a common sanity checker component is useful. And I think that can be a part of Amit's project. Create a backlog issue to track this.

Reply:
> why allow user to create option?

This change is not a final solution, just some workaround to support remote cluster indices. 
For local indices, user still can select filed from dropdown list. For remote cluster indices, ES doesn't support listing remote cluster's indices, not get index mapping. We can't get remote cluster index's fields. So for remote cluster indices, user need to input filed manually.

Reply:
Thanks. Btw, why timestamp field from remote index is not visible?

Reply:
> Thanks. Btw, why timestamp field from remote index is not visible?

For local index, we can get its index mapping and parse which filed is timestamp type. But for remote cluster index, we can't get its index mapping. So we don't know which filed is timestamp.